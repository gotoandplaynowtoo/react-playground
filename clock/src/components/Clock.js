import React from 'react';
import PropTypes from 'prop-types';

const Clock = (props) => (
    <div style={{
        color: '#3498db',
        fontFamily: 'monospace',
        fontSize: 24,
        fontWeight: 'bold',
    }}>
        {props.dateString}
    </div>
);

Clock.propTypes = {
    dateString: PropTypes.string
};

Clock.defaultProps = {
    dateString: 'Clock default string'
};

export default Clock;