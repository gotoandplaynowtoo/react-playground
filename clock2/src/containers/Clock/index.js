import React, { Component } from 'react';
import Display from 'components/Display';

class Clock extends Component {

  constructor(props) {
    super(props);
    this.state = {
      time: ''
    };
    this.timerId = null;
  }

  componentDidMount() {
    this.timerId = setInterval(() => {
      this.tick();
    }, 1000);
  }

  componentWillUnmount() {
    clearInterval(this.timerId);
  }

  render() {
    return <Display>{this.state.time}</Display>
  }

  tick() {
    this.setState({
      time: (new Date()).toLocaleTimeString()
    });
  }

}

export default Clock;