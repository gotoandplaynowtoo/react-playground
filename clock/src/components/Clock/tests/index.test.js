import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Clock from 'components/Clock';

it('render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Clock/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('render correctly', () => {
  const tree = renderer.create(<Clock/>).toJSON();
  expect(tree).toMatchSnapshot();
});
