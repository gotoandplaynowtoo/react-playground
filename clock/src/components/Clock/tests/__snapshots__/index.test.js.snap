// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div
  style={
    Object {
      "color": "#3498db",
      "fontFamily": "monospace",
      "fontSize": 24,
      "fontWeight": "bold",
    }
  }
>
  Clock default string
</div>
`;
